AWSAuthSDK/Sources/AWSMobileClient/Helpers/UserPoolHelper/AWSMobileClient+UserPoolAuthHelperCallbacks.swift

```diff
@@ -49,7 +49,9 @@ extension AWSMobileClient: UserPoolAuthHelperCallbacks {
     func getNewPasswordDetails(_ newPasswordRequiredInput: AWSCognitoIdentityNewPasswordRequiredInput,
                                newPasswordRequiredCompletionSource: AWSTaskCompletionSource<AWSCognitoIdentityNewPasswordRequiredDetails>) {
         self.userpoolOpsHelper.newPasswordRequiredTaskCompletionSource = newPasswordRequiredCompletionSource
-        let result = SignInResult(signInState: .newPasswordRequired, codeDetails: nil)
+        let result = SignInResult(signInState: .newPasswordRequired,
+                                  parameters: ["requiredAttributes": newPasswordRequiredInput.requiredAttributes.joined(separator: " ")],
+                                  codeDetails: nil)
```

Comment:
I would suggest to make the following change instead, so that it can be be decoded into different types. 

```suggestion
        let result = SignInResult(
            signInState: .newPasswordRequired,
            parameters: [
                "requiredAttributes": newPasswordRequiredInput.requiredAttributes.description,
                "userAttributes": newPasswordRequiredInput.userAttributes.description
            ]
        )
```